src/test/kotlin/ecommerce/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,74 @@
+package ecommerce.controller
+
+import ecommerce.model.Product
+import io.restassured.RestAssured
+import io.restassured.http.ContentType
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.test.annotation.DirtiesContext
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class ProductControllerTest {
+    private var productId: Long = -1
+
+    @BeforeEach
+    fun createProducts() {
+        val product = Product(0, "cafe", 39.0, "www.test")
+        val response =
+            RestAssured
+                .given().log().all().body(product)
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().extract().response()
+
+        productId = response.jsonPath().getLong("id")
+    }
+
+    @Test
+    fun addProduct() {
+        val product = Product(0, "table", 45.0, "test.com")
+        val response =
+            RestAssured
+                .given().log().all().body(product)
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().extract().response()
+
+        assertThat(response.statusCode).isEqualTo(HttpStatus.OK.value())
+        val created = response.`as`(Product::class.java)
+        assertThat(created.name).isEqualTo("table")
+        assertThat(created.price).isEqualTo(45.0)
+        assertThat(created.imageUrl).isEqualTo("test.com")
```

Comment:
```suggestion
        RestAssured
            .given().log().all().body(product)
            .contentType(ContentType.JSON)
            .`when`().post("/api/products")
            .then()
            .statusCode(HttpStatus.OK.value())
            .body("name", equalTo("table"))
            .body("price", equalTo(45.0F))
            .body("imageUrl", equalTo("test.com"))
```
This is not a must, but we can also let `RestAssured` assert everything :)